src/gui/filedetails/sharemodel.cpp

```diff
@@ -17,27 +17,82 @@
 #include <QFileInfo>
 #include <QTimeZone>
 
+#include <random>
+#include <openssl/rand.h>
+
 #include "account.h"
 #include "folderman.h"
 #include "theme.h"
-#include "wordlist.h"
 
 namespace {
 
 static const auto placeholderLinkShareId = QStringLiteral("__placeholderLinkShareId__");
 static const auto internalLinkShareId = QStringLiteral("__internalLinkShareId__");
 static const auto secureFileDropPlaceholderLinkShareId = QStringLiteral("__secureFileDropPlaceholderLinkShareId__");
 
+constexpr auto asciiMin = 33;
+constexpr auto asciiMax = 126;
+constexpr auto asciiRange = asciiMax - asciiMin;
+
 QString createRandomPassword()
 {
-    const auto words = OCC::WordList::getRandomWords(10);
+    static constexpr auto numChars = 24;
+
+    static constexpr std::string_view lowercaseAlphabet = "abcdefghijklmnopqrstuvwxyz";
+    static constexpr std::string_view uppercaseAlphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    static constexpr std::string_view numbers = "0123456789";
+    static constexpr std::string_view specialChars = R"(ªº\\/|"'*+-_´¨{}·#$%&()=\[\]<>;:@~)";
 
-    const auto addFirstLetter = [](const QString &current, const QString &next) -> QString {
-        return current + next.at(0);
+    static const QRegularExpression lowercaseMatch("[a-z]");
+    static const QRegularExpression uppercaseMatch("[A-Z]");
+    static const QRegularExpression numberMatch("[0-9]");
+    static const QRegularExpression specialCharMatch(QString("[%1]").arg(specialChars.data()));
+
+    static const std::map<std::string_view, QRegularExpression> matchMap {
+        { lowercaseAlphabet, lowercaseMatch },
+        { uppercaseAlphabet, uppercaseMatch },
+        { numbers, numberMatch },
+        { specialChars, specialCharMatch },
     };
 
-    return std::accumulate(std::cbegin(words), std::cend(words), QString(), addFirstLetter);
+    std::random_device rand_dev;
+    std::mt19937 rng(rand_dev());
+
+    QString passwd;
+    unsigned char unsignedCharArray[numChars];
+
+    RAND_bytes(unsignedCharArray, numChars);
+
+    for (auto i = 0; i < numChars; i++) {
```

Comment:
`for (const auto newChar : unsignedCharArray) {`

Reply:
Changes this and simplified it